src/iwant_bot/iwant_process.py

```diff
@@ -0,0 +1,116 @@
+from dateparser import parse
+from datetime import datetime, timedelta
+import re
+
+
+class IwantRequest(object):
+    """Valid iwant command create this object, which extracts information with /iwant requests.
+    Other_words are expected words in the user message, which should be removed before dateparse."""
+
+    def __init__(self, body: dict, possible_activities: tuple, possible_behests: tuple,
+                 user_pattern: str, other_words: tuple = (), default_duration: float = 900.0,
+                 max_duration: float = None):
+
+        self.possible_activities = possible_activities
+        self.possible_behests = possible_behests
+        self.default_duration = default_duration
+        self.max_duration = max_duration
+
+        self.data = body
+        self.data['activities'] = parse_text_for_words(body['text'], possible_activities)
+        self.data['behests'] = parse_text_for_words(body['text'], possible_behests)
+        self.data['invite_users_id'] = parse_text_for_people(body['text'], user_pattern)
+        date = parse_text_for_time(
+            body['text'], user_pattern, possible_activities + possible_behests + other_words)
+
+        if date is None:
+            date = datetime.now() + timedelta(seconds=default_duration)
+
+        self.data['deadline'] = date
+        self.data['action_start'] = datetime.now()
+        self.data['action_duration'] = (self.data['deadline']
+                                        - self.data['action_start']).total_seconds()
+        if max_duration is not None:
+            self.data['action_duration'] = min(self.data['action_duration'], max_duration)
+        if 'callback_id' not in self.data:
+            self.data['callback_id'] = []
+
+    def return_list_of_parameters(self) -> dict:
```

Comment:
I recommend this restructuring: Define variables e.g. `possible_activities` and `possible_behests` and format everything in a long format string - this would more clearly separate the formatting of information from gathering of information. This is also nice e.g. for translators.